cli/diagnostics.rs

```diff
@@ -0,0 +1,754 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+//! This mod adds source maps and ANSI color display to deno::DenoDiagnostic.
+use crate::ansi;
+use deno;
+use deno::DenoDiagnostic;
+use deno::DenoDiagnosticCategory;
+use deno::DenoDiagnosticFrame;
+use deno::DenoDiagnosticSources;
+use source_map_mappings::parse_mappings;
+use source_map_mappings::Bias;
+use source_map_mappings::Mappings;
+use std::collections::HashMap;
+use std::fmt;
+use std::str;
+
+/// Wrapper around DenoDiagnostic which provides color to_string.
+pub struct DenoDiagnosticColor<'a>(pub &'a DenoDiagnostic);
+
+struct DenoDiagnosticFrameColor<'a>(&'a DenoDiagnosticFrame);
+
+pub trait SourceMapGetter {
+  /// Returns the raw source map file.
+  fn get_source_map(&self, script_name: &str) -> Option<Vec<u8>>;
+}
+
+/// Cached filename lookups. The key can be None if a previous lookup failed to
+/// find a SourceMap.
+type CachedMaps = HashMap<String, Option<SourceMap>>;
+
+struct SourceMap {
+  mappings: Mappings,
+  sources: Vec<String>,
+}
+
+impl<'a> fmt::Display for DenoDiagnosticFrameColor<'a> {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    let frame = self.0;
+    // Note when we print to string, we change from 0-indexed to 1-indexed.
+    let function_name = ansi::italic_bold(frame.function_name.clone());
+    let script_line_column =
+      format_script_line_column(&frame.script_name, frame.line, frame.column);
+
+    if !frame.function_name.is_empty() {
+      write!(f, "    at {} ({})", function_name, script_line_column)
+    } else if frame.is_eval {
+      write!(f, "    at eval ({})", script_line_column)
+    } else {
+      write!(f, "    at {}", script_line_column)
+    }
+  }
+}
+
+fn format_script_line_column(
+  script_name: &str,
+  line: i64,
+  column: i64,
+) -> String {
+  // TODO match this style with how typescript displays errors.
+  let line = ansi::yellow((1 + line).to_string());
```

Comment:
These should already match TS colors, I think I did in in #1793

Reply:
This was a comment copied over from other code, will eliminate.